#Trendy package
```{r}
if (!requireNamespace("BiocManager", quietly=TRUE))
  install.packages("BiocManager")
BiocManager::install("Trendy")

library(Trendy)
```

#Data processing for TRENDY
```{r}
library("dplyr")
library("tidyr")
mRNAseqData=read.table("~/Desktop/gcloud_2021_06/rstudio_2021_06/mRNA-seq_hg38.gencode21.wholeGene.geneComposite.STAR.nochrM.gene.count.txt",sep="\t",header=TRUE)
modsb1= mRNAseqData %>% 
  separate(Geneid,c("EnsemblID","Genename"),extra="merge")
modsb1$EnsemblID<-NULL
#Filtering for Akey alone
akeySestan1 <- modsb1 %>% filter(modsb1$Genename %in% results$hgnc_symbol) #From jm_Allen.R
akeySestan1=t(akeySestan1)

#As dataframe
akeySestan1=as.data.frame(akeySestan1)

colnames(akeySestan1) <- as.matrix(unlist(akeySestan1[1,]))
akeySestan1 <- akeySestan1[-1, ]

akeySestan1 <- cbind(info = rownames(akeySestan1), akeySestan1)
rownames(akeySestan1) <- 1:nrow(akeySestan1)

#duplicated columns - issue in raw data. Here: 
colnames(akeySestan1)[duplicated(colnames(akeySestan1))] #1
akeySestan1 <- akeySestan1[, !duplicated(colnames(akeySestan1))]

akeySestan1=akeySestan1 %>% separate(info, c("Braincode","Regioncode"))


#Brining the metadata of the database
library(readxl)
metadatamRNAseq=read_xlsx("~/Desktop/gcloud_2021_06/rstudio_2021_06/tmp_psychENCODE/mRNA-seq_QC.xlsx",skip = 3)
modMetadatamRNAseq=na.omit(metadatamRNAseq)
modMetadatamRNAseq=modMetadatamRNAseq %>% dplyr::select(1:3)
modMetadatamRNAseq=as.data.frame(modMetadatamRNAseq)

finalakeySestan1=merge(modMetadatamRNAseq,akeySestan1,by=c("Braincode", "Regioncode"))



#Pey And Deserts
pey_coords <- read.delim("~/Desktop/gcloud_2021_06/rstudio_2021_06/2020_pey_coords.bed", header=FALSE) #File with start 0-based
###Preparing bed file for input in bioMart
pey_coords$V1 <- gsub("chr", "\\1", pey_coords$V1)
pey_coords[2] <- pey_coords[2]+1 #Moving to 1-based for Ensembl
df <- paste(pey_coords$V1, pey_coords$V2, pey_coords$V3, sep = ":")
results_pey=getBM(attributes = c("hgnc_symbol", "chromosome_name", "start_position", "end_position","gene_biotype"),
              filters = c("chromosomal_region","biotype"),
              values = list(chromosomal_region=df,biotype="protein_coding"), mart = ensembl)
results_pey <- results_pey[!duplicated(results_pey$hgnc_symbol),]
###Genes within Akey and Pey - RESULT
both <- results_pey[results_pey$hgnc_symbol %in% results$hgnc_symbol,]
both <-  both[!(is.na(both$hgnc_symbol) | both$hgnc_symbol==""), ]
```

#CEREBELLUM
```{r}
cbcakey <- filter(finalakeySestan1, Regioncode == "CBC")
cbcakey <- arrange(cbcakey, Window)
time.vector <- cbcakey$Window
cbcakey <- cbcakey %>% unite("code", c("Braincode","Regioncode","Window"))
rownames(cbcakey) <- cbcakey$code
cbcakey$code <- NULL

cbcakey1 <- as_tibble(cbcakey)

cols.num <- colnames(cbcakey1)
cbcakey1[,cols.num] <- lapply(cbcakey1[cols.num],as.character)
cbcakey1[,cols.num] <- lapply(cbcakey1[cols.num],as.numeric)

finalcbc <- t(cbcakey1)
colnames(finalcbc) <- time.vector 
set.seed(10)
res <- trendy(Data = finalcbc, tVectIn = time.vector, maxK = 3, minNumInSeg = 2, meanCut = 2)
res <- results(res)
res.top <- topTrendy(res)
#Good fit in Des & Pos sel.
rownames(res.top$Trends)[rownames(res.top$Trends) %in% both$hgnc_symbol]

res.trend <- trendHeatmap(res.top)

rcbc <- as.data.frame(res.top$AdjustedR2)
names(rcbc) <- "Cerebellar cortex - R2"
write.csv(rcbc, file="~/desertsHomo/1.data/trendy_adjR2/CBC_GenesDesertPosSel_adjR2.csv", row.names = TRUE, quote=T)

##Heatmap homemade
class(res.top$Trends)
pheatmap(res.top$Trends, cluster_rows=FALSE, cluster_cols = FALSE, legend_breaks = c(-1,0,1), filename = "~/desertsHomo/2.plots/trendy_segmentedRegression/CBC_UpDown_Deserts.pdf")

sb_inDesPey <- res.top$Trends[rownames(res.top$Trends)[rownames(res.top$Trends) %in% both$hgnc_symbol],]

pheatmap(sb_inDesPey, cluster_rows=FALSE, cluster_cols = FALSE, legend_breaks = c(-1,0,1),filename = "~/desertsHomo/2.plots/trendy_segmentedRegression/CBC_UpDown_DesertsPosSel.pdf")


#Selected genes
par(mfrow=c(3,2))
plot2 <- plotFeature(finalcbc,tVectIn = time.vector,featureNames = c("SYT6", "ROBO2", "CADPS2", "GPR22", "BCAP29"),trendyOutData = res)

pdf("~/desertsHomo/2.plots/trendy_segmentedRegression/CBC_GenesDesertPosSel.pdf")
par(mfrow=c(3,2))
plot2 <- plotFeature(finalcbc,tVectIn = time.vector,featureNames = c("SYT6", "ROBO2", "CADPS2", "GPR22", "BCAP29"),trendyOutData = res, customLabels.x = c("Fetal_1","Fetal_3","Child","Adult" ), xlab="")
dev.off()
```

#STRIATUM
```{r}
strakey <- filter(finalakeySestan1, Regioncode == "STR")
strakey <- arrange(strakey, Window)
time.vector <- strakey$Window
strakey <- strakey %>% unite("code", c("Braincode","Regioncode","Window"))
rownames(strakey) <- strakey$code
strakey$code <- NULL

strakey1 <- as_tibble(strakey)

cols.num <- colnames(strakey1)
strakey1[,cols.num] <- lapply(strakey1[cols.num],as.character)
strakey1[,cols.num] <- lapply(strakey1[cols.num],as.numeric)

finalstr <- t(strakey1)
colnames(finalstr) <- time.vector 
set.seed(10)
res <- trendy(Data = finalstr, tVectIn = time.vector, maxK = 3, minNumInSeg = 2, meanCut = 2)
res <- results(res)
res.top <- topTrendy(res)

rstr <- as.data.frame(res.top$AdjustedR2)
names(rstr) <- "Striatum - R2"
write.csv(rstr, file="~/desertsHomo/1.data/trendy_adjR2/STR_GenesDesertPosSel_adjR2.csv", row.names = TRUE, quote=T)

#Good fit in Des & Pos sel.
rownames(res.top$Trends)[rownames(res.top$Trends) %in% both$hgnc_symbol]

res.trend <- trendHeatmap(res.top)
##Heatmap homemad
class(res.top$Trends)
pheatmap(res.top$Trends, cluster_rows=FALSE, cluster_cols = FALSE, legend_breaks = c(-1,0,1), filename = "~/desertsHomo/2.plots/trendy_segmentedRegression/STR_UpDown_Deserts.pdf")

sb_inDesPey <- res.top$Trends[rownames(res.top$Trends)[rownames(res.top$Trends) %in% both$hgnc_symbol],]

pheatmap(sb_inDesPey, cluster_rows=FALSE, cluster_cols = FALSE, legend_breaks = c(-1,0,1),filename = "~/desertsHomo/2.plots/trendy_segmentedRegression/STR_UpDown_DesertsPosSel.pdf")


#Selected genes
par(mfrow=c(2,2))
plot2 <- plotFeature(finalstr,tVectIn = time.vector,featureNames = c("ROBO2", "BCAP29", "ST7"),trendyOutData = res)

pdf("~/desertsHomo/2.plots/trendy_segmentedRegression/STR_GenesDesertPosSel.pdf")
par(mfrow=c(3,2))
plot2 <- plotFeature(finalstr,tVectIn = time.vector,featureNames = c("ROBO2", "BCAP29", "ST7", "SYT6"),trendyOutData = res, customLabels.x = c("Fetal_1","Fetal_3","Child"), xlab="")
dev.off()
```

#THALAMUS
```{r}
mdtakey <- filter(finalakeySestan1, Regioncode == "MD")
mdtakey <- arrange(mdtakey, Window)
time.vector <- mdtakey$Window
mdtakey <- mdtakey %>% unite("code", c("Braincode","Regioncode","Window"))
rownames(mdtakey) <- mdtakey$code
mdtakey$code <- NULL

mdtakey1 <- as_tibble(mdtakey)

cols.num <- colnames(mdtakey1)
mdtakey1[,cols.num] <- lapply(mdtakey1[cols.num],as.character)
mdtakey1[,cols.num] <- lapply(mdtakey1[cols.num],as.numeric)

finalmdt <- t(mdtakey1)
colnames(finalmdt) <- time.vector 
set.seed(10)
res <- trendy(Data = finalmdt, tVectIn = time.vector, maxK = 3, minNumInSeg = 2, meanCut = 2)
res <- results(res)
res.top <- topTrendy(res)

rmdt <- as.data.frame(res.top$AdjustedR2)
names(rmdt) <- "MD - R2"
#write.csv(rmdt, file="~/desertsHomo/1.data/trendy_adjR2/MDT_GenesDesertPosSel_adjR2.csv", row.names = TRUE, quote=T)

#Good fit in Des & Pos sel.
rownames(res.top$Trends)[rownames(res.top$Trends) %in% both$hgnc_symbol]

res.trend <- trendHeatmap(res.top)
##Heatmap homemade
class(res.top$Trends)
library(pheatmap)
pheatmap(res.top$Trends, cluster_rows=FALSE, cluster_cols = FALSE, legend_breaks = c(-1,0,1), filename = "~/desertsHomo/2.plots/trendy_segmentedRegression/MDT_UpDown_Deserts.pdf")

sb_inDesPey <- res.top$Trends[rownames(res.top$Trends)[rownames(res.top$Trends) %in% both$hgnc_symbol],]

pheatmap(sb_inDesPey, cluster_rows=FALSE, cluster_cols = FALSE, legend_breaks = c(-1,0,1),filename = "~/desertsHomo/2.plots/trendy_segmentedRegression/MDT_UpDown_DesertsPosSel.pdf")


#Selected genes
par(mfrow=c(2,2))
plot2 <- plotFeature(finalmdt,tVectIn = time.vector,featureNames = c("ROBO2", "ST7"),trendyOutData = res)

pdf("~/desertsHomo/2.plots/trendy_segmentedRegression/MDT_GenesDesertPosSel.pdf")
par(mfrow=c(3,2))
plot2 <- plotFeature(finalmdt,tVectIn = time.vector,featureNames = c("ROBO2", "ST7"),trendyOutData = res, customLabels.x = c("Fetal_1","Fetal_3","Child"), xlab="")
dev.off()
```

#ABADATA
```{r}
if (!requireNamespace("BiocManager", quietly = TRUE))
    install.packages("BiocManager")

BiocManager::install("ABAData")
library(ABAData)
BiocManager::install("ABAEnrichment")
library(ABAEnrichment)


##Loading dataset
data("dataset_5_stages")
#unique(dataset_5_stages$structure) #Checking structures

#Selecting genes ID and structures present in dataset
id <- unique(dataset_5_stages$ensembl_gene_id)
st <- unique(dataset_5_stages$structure)
st_allen <- paste("Allen",st, sep=":") 
#Expression data for all structures and genes
ab <- get_expression(structure_ids=st_allen, gene_ids = id, dataset='5_stages')

#Converting data to a dataframe with useful format for later
list1 = vector(mode="list")
for (r in 1:length(ab)){
  ab[[r]] <- t(ab[[r]]) #transpose
  list1 <- get_name(colnames(ab[[r]])) #change Allen:XXXX to e.g. M1C_primary motor cortex, etc
  colnames(ab[[r]]) <- list1
  ab[[r]] <- as.data.frame(ab[[r]])
}

ab1 = vector(mode="list", length = length(ab))
 for (i in 1:length(ab)){
   for (h in 1:length(names(ab[[i]]))){
     ab1[[i]][[h]] <- ab[[i]][h]
   }
 }
 
for (i in 1:length(ab)){
   for (h in 1:length(names(ab[[i]]))){
     G_list <- getBM(filters= "ensembl_gene_id", attributes= c("ensembl_gene_id", "hgnc_symbol"),values=rownames(ab1[[i]][[h]]),mart=ensembl)
     ab1[[i]][[h]]['gene_name'] <-  G_list$hgnc_symbol[match(rownames(ab1[[i]][[h]]), G_list$ensembl_gene_id)]
     ab1[[i]][[h]] <-ab1[[i]][[h]][order(ab1[[i]][[h]][[1]], decreasing = TRUE), ]
   }
}


#CEREBELLUM
ab_t <- as.data.frame(cbind(ab1[[1]][[16]], ab1[[2]][[16]], ab1[[3]][[16]], ab1[[4]][[16]], ab1[[5]][[16]]))
colnames(ab_t) <- c("CBC_1", "gene_name","CBC_2", "gene_name","CBC_3", "gene_name","CBC_4","gene_name", "CBC_5","gene_name")
rownames(ab_t) <- NULL


ab_1 <- as.data.frame(cbind(ab1[[1]][[16]]))
ab_1 <- ab_1[!(is.na(ab_1$gene_name) | ab_1$gene_name==""), ]
rownames(ab_1) <- NULL
colnames(ab_1)[1] <- "CBC_1"

ab_2 <- as.data.frame(cbind(ab1[[2]][[16]]))
ab_2 <- ab_2[!(is.na(ab_2$gene_name) | ab_2$gene_name==""), ]
rownames(ab_2) <- NULL
colnames(ab_2)[1] <- "CBC_2"

ab_3 <- as.data.frame(cbind(ab1[[3]][[16]]))
ab_3 <- ab_3[!(is.na(ab_3$gene_name) | ab_3$gene_name==""), ]
rownames(ab_3) <- NULL
colnames(ab_3)[1] <- "CBC_3"

ab_4 <- as.data.frame(cbind(ab1[[4]][[16]]))
ab_4 <- ab_4[!(is.na(ab_4$gene_name) | ab_4$gene_name==""), ]
rownames(ab_4) <- NULL
colnames(ab_4)[1] <- "CBC_4"


ab_5 <- as.data.frame(cbind(ab1[[5]][[16]]))
ab_5 <- ab_5[!(is.na(ab_5$gene_name) | ab_5$gene_name==""), ]
rownames(ab_5) <- NULL
colnames(ab_5)[1] <- "CBC_5"

aabb  <- right_join(ab_1, ab_2, by = "gene_name")
aabb  <- right_join(aabb, ab_3, by = "gene_name")
aabb  <- right_join(aabb, ab_4, by = "gene_name")
aabb  <- right_join(aabb, ab_5, by = "gene_name")
aabb <- aabb[!(is.na(aabb$gene_name) | aabb$gene_name==""), ]
rownames(aabb) <- aabb$gene_name
aabb$gene_name<- NULL

time.vector <- c(rep(1:5))

colnames(aabb) <- time.vector 
set.seed(10)
res <- trendy(Data = aabb, tVectIn = time.vector, maxK = 2, minNumInSeg = 2, meanCut = 2)
res_alt <- trendy(Data = aabb, tVectIn = time.vector, maxK = 2, minNumInSeg = 1, meanCut = 2)


res <- results(res)
res.top <- topTrendy(res)

res_alt <- results(res_alt)
res_alt.top <- topTrendy(res_alt)

#Good fit in Des & Pos sel.
rownames(res.top$Trends)[rownames(res.top$Trends) %in% both$hgnc_symbol]
rownames(res_alt.top$Trends)[rownames(res_alt.top$Trends) %in% both$hgnc_symbol]


res.trend <- trendHeatmap(res.top)
res_alt.trend <- trendHeatmap(res_alt.top)

rcbc_aba <- as.data.frame(res.top$AdjustedR2)
names(rcbc_aba) <- "Cerebellar cortex - R2"
write.csv(rcbc_aba, file="~/desertsHomo/1.data/trendy_adjR2/CBC_GenesDesertPosSel_adjR2_ABA.csv", row.names = TRUE, quote=T)

##Heatmap homemade
library(pheatmap)
class(res.top$Trends)
pheatmap(res.top$Trends, cluster_rows=FALSE, cluster_cols = FALSE, legend_breaks = c(-1,0,1), filename = "~/desertsHomo/2.plots/trendy_segmentedRegression/CBC_UpDown_Deserts.pdf")

sb_inDesPey <- res.top$Trends[rownames(res.top$Trends)[rownames(res.top$Trends) %in% both$hgnc_symbol],]

pheatmap(sb_inDesPey, cluster_rows=FALSE, cluster_cols = FALSE, legend_breaks = c(-1,0,1),filename = "~/desertsHomo/2.plots/trendy_segmentedRegression/CBC_UpDown_DesertsPosSel_ABA.pdf")


#Selected genes
par(mfrow=c(3,2))
pdf("~/CBC_GenesDesertsPosSel_ABA.pdf")
plot2 <- plotFeature(aabb,tVectIn = time.vector,featureNames = c("ST7", "ROBO2", "CADPS2", "GPR22", "BCAP29", "KCND2", "SLC16A1"),trendyOutData = res)
dev.off()


par(mfrow=c(3,2))
pdf("~/CBC_GenesDesertsPosSel_ABA_alt.pdf")
plot2 <- plotFeature(aabb,tVectIn = time.vector,featureNames = c("ST7", "ROBO2", "CADPS2", "GPR22", "BCAP29", "KCND2", "SLC16A1"),trendyOutData = res_alt)
dev.off()






par(mfrow=c(3,2))
plot2 <- plotFeature(finalcbc,tVectIn = time.vector,featureNames = c("SYT6", "ROBO2", "CADPS2", "GPR22", "BCAP29"),trendyOutData = res, customLabels.x = c("Fetal_1","Fetal_3","Child","Adult" ), xlab="")
dev.off()



#THALAMUS
ab_t <- as.data.frame(cbind(ab1[[1]][[15]], ab1[[2]][[15]], ab1[[3]][[15]], ab1[[4]][[15]], ab1[[5]][[15]]))
colnames(ab_t) <- c("MD_1", "gene_name","MD_2", "gene_name","MD_3", "gene_name","MD_4","gene_name", "MD_5","gene_name")
rownames(ab_t) <- NULL


ab_1 <- as.data.frame(cbind(ab1[[1]][[15]]))
ab_1 <- ab_1[!(is.na(ab_1$gene_name) | ab_1$gene_name==""), ]
rownames(ab_1) <- NULL
colnames(ab_1)[1] <- "MD_1"

ab_2 <- as.data.frame(cbind(ab1[[2]][[15]]))
ab_2 <- ab_2[!(is.na(ab_2$gene_name) | ab_2$gene_name==""), ]
rownames(ab_2) <- NULL
colnames(ab_2)[1] <- "MD_2"

ab_3 <- as.data.frame(cbind(ab1[[3]][[15]]))
ab_3 <- ab_3[!(is.na(ab_3$gene_name) | ab_3$gene_name==""), ]
rownames(ab_3) <- NULL
colnames(ab_3)[1] <- "MD_3"

ab_4 <- as.data.frame(cbind(ab1[[4]][[15]]))
ab_4 <- ab_4[!(is.na(ab_4$gene_name) | ab_4$gene_name==""), ]
rownames(ab_4) <- NULL
colnames(ab_4)[1] <- "MD_4"


ab_5 <- as.data.frame(cbind(ab1[[5]][[15]]))
ab_5 <- ab_5[!(is.na(ab_5$gene_name) | ab_5$gene_name==""), ]
rownames(ab_5) <- NULL
colnames(ab_5)[1] <- "MD_5"

aabb  <- right_join(ab_1, ab_2, by = "gene_name")
aabb  <- right_join(aabb, ab_3, by = "gene_name")
aabb  <- right_join(aabb, ab_4, by = "gene_name")
aabb  <- right_join(aabb, ab_5, by = "gene_name")
aabb <- aabb[!(is.na(aabb$gene_name) | aabb$gene_name==""), ]
rownames(aabb) <- aabb$gene_name
aabb$gene_name<- NULL

time.vector <- c(rep(1:5))

colnames(aabb) <- time.vector 
set.seed(10)
res <- trendy(Data = aabb, tVectIn = time.vector, maxK = 2, minNumInSeg = 2, meanCut = 2)
#res_alt <- trendy(Data = aabb, tVectIn = time.vector, maxK = 2, minNumInSeg = 1, meanCut = 2)


res <- results(res)
res.top <- topTrendy(res)

#res_alt <- results(res_alt)
#res_alt.top <- topTrendy(res_alt)

#Good fit in Des & Pos sel.
rownames(res.top$Trends)[rownames(res.top$Trends) %in% both$hgnc_symbol]
#rownames(res_alt.top$Trends)[rownames(res_alt.top$Trends) %in% both$hgnc_symbol]


res.trend <- trendHeatmap(res.top)
#res_alt.trend <- trendHeatmap(res_alt.top)

rMD_aba <- as.data.frame(res.top$AdjustedR2)
names(rMD_aba) <- "MD Thalamus - R2"
write.csv(rMD_aba, file="~/desertsHomo/1.data/trendy_adjR2/MD_GenesDesertPosSel_adjR2_ABA.csv", row.names = TRUE, quote=T)

##Heatmap homemade
library(pheatmap)
class(res.top$Trends)
pheatmap(res.top$Trends, cluster_rows=FALSE, cluster_cols = FALSE, legend_breaks = c(-1,0,1), filename = "~/desertsHomo/2.plots/trendy_segmentedRegression/MDT_UpDown_Deserts_ABA.pdf")

sb_inDesPey <- res.top$Trends[rownames(res.top$Trends)[rownames(res.top$Trends) %in% both$hgnc_symbol],]

pheatmap(sb_inDesPey, cluster_rows=FALSE, cluster_cols = FALSE, legend_breaks = c(-1,0,1),filename = "~/desertsHomo/2.plots/trendy_segmentedRegression/MDT_UpDown_DesertsPosSel_ABA.pdf")


#Selected genes
par(mfrow=c(2,2))
pdf("~/desertsHomo/2.plots/trendy_segmentedRegression/MDT_GenesDesertsPosSel_ABA.pdf")
plot2 <- plotFeature(aabb,tVectIn = time.vector,featureNames = c("SYT6", "ROBO2", "SLC16A1"),trendyOutData = res)
dev.off()


#par(mfrow=c(3,2))
#pdf("~/MDT_GenesDesertsPosSel_ABA_alt.pdf")
#plot2 <- plotFeature(aabb,tVectIn = time.vector,featureNames = c("ST7", "ROBO2", "CADPS2", "GPR22", "BCAP29", "KCND2", "SLC15A1"),trendyOutData = res_alt)
#dev.off()
```

