#Packages
```{r}
#Packages
library(biomaRt)
library(ABAData)
library(ABAEnrichment)
library(data.table)
library(ggplot2)
library(dplyr)
library(GGally)
library(viridis)
library(ggpubr)
library(grid)
library(gridExtra)
library(lattice)
library(xlsx)
library(tidyr)
library(tidyverse)
library(DescTools)
library(reshape2)
library(dplyr)
library(Matrix)
library(pheatmap)
library(plyr)
library(rstatix)
library(matrixStats)
```

#Extracting gene names from coordinate regions via bioMart
```{r}
#Genes within Desert region coordinates
##Using hg19 genome
ensembl <- useMart(biomart = 'ENSEMBL_MART_ENSEMBL', 
                        dataset = 'hsapiens_gene_ensembl',
                        host = 'https://grch37.ensembl.org')
#Total number of protein coding genes in Ensembl
resP=getBM(attributes = c("hgnc_symbol","gene_biotype"),
              filters = "biotype",
              values = list(biotype="protein_coding"), mart = ensembl)

length(resP$hgnc_symbol[!(is.na(resP$hgnc_symbol) | resP$hgnc_symbol=="")])

##Akey Deserts coordinates (Ensembl 1-based):
filterlist <- c("1:105400000:120600000", "3:74100000:89300000", "7:106200000:123200000","8:49400000:66500000")
##Select only protein-coding from Akey
results=getBM(attributes = c("hgnc_symbol", "chromosome_name", "start_position", "end_position","gene_biotype"),
              filters = c("chromosomal_region","biotype"),
              values = list(chromosomal_region=filterlist,biotype="protein_coding"), mart = ensembl)
results <- results[!duplicated(results$hgnc_symbol),]
#255
length(results$hgnc_symbol[!(is.na(results$hgnc_symbol) | results$hgnc_symbol=="")])
results <-  results[!(is.na(results$hgnc_symbol) | results$hgnc_symbol==""), ] #Cleaning

##Pey coordinates (Ensembl 1-based):
pey_coords <- read.delim("~/Desktop/gcloud_2021_06/rstudio_2021_06/2020_pey_coords.bed", header=FALSE) #File with start 0-based
###Preparing bed file for input in bioMart
pey_coords$V1 <- gsub("chr", "\\1", pey_coords$V1)
pey_coords[2] <- pey_coords[2]+1 #Moving to 1-based for Ensembl
df <- paste(pey_coords$V1, pey_coords$V2, pey_coords$V3, sep = ":")
results_pey=getBM(attributes = c("hgnc_symbol", "chromosome_name", "start_position", "end_position","gene_biotype"),
              filters = c("chromosomal_region","biotype"),
              values = list(chromosomal_region=df,biotype="protein_coding"), mart = ensembl)
results_pey <- results_pey[!duplicated(results_pey$hgnc_symbol),]
###Genes within Akey and Pey - RESULT
both <- results_pey[results_pey$hgnc_symbol %in% results$hgnc_symbol,]
both <-  both[!(is.na(both$hgnc_symbol) | both$hgnc_symbol==""), ] #Cleaning

###Rac coordinates #MOVE TO ENSEMBL 1 BASED
rac_coords <- read.delim("~/2020_rac_coords.bed", header=FALSE)
rac_coords$V1 <- gsub("chr", "\\1", rac_coords$V1)
rac_coords[2] <- rac_coords[2]+1

df1 <- paste(rac_coords$V1, rac_coords$V2, rac_coords$V3, sep = ":")
results_rac=getBM(attributes = c("hgnc_symbol", "chromosome_name", "start_position", "end_position","gene_biotype"),
              filters = c("chromosomal_region","biotype"),
              values = list(chromosomal_region=df1,biotype="protein_coding"), mart = ensembl)
results_rac <- results_rac[!duplicated(results_rac$hgnc_symbol),]
###Genes within Akey and Rac - RESULT
racAkey <- results_rac[results_rac$hgnc_symbol %in% results$hgnc_symbol,]
racAkey <- racAkey[!(is.na(racAkey$hgnc_symbol) | racAkey$hgnc_symbol==""), ]
```

#Extracting gene expression data from Allen Brain Atlas - 5 stages
```{r}
##Loading dataset
data("dataset_5_stages")
#unique(dataset_5_stages$structure) #Checking structures

#Selecting genes ID and structures present in dataset
id <- unique(dataset_5_stages$ensembl_gene_id)
st <- unique(dataset_5_stages$structure)
st_allen <- paste("Allen",st, sep=":") 
#Expression data for all structures and genes
ab <- get_expression(structure_ids=st_allen, gene_ids = id, dataset='5_stages')

#Converting data to a dataframe with useful format for later
list1 = vector(mode="list")
for (r in 1:length(ab)){
  ab[[r]] <- t(ab[[r]]) #transpose
  list1 <- get_name(colnames(ab[[r]])) #change Allen:XXXX to e.g. M1C_primary motor cortex, etc
  colnames(ab[[r]]) <- list1
  ab[[r]] <- as.data.frame(ab[[r]])
}
```

#ABA Data - All genes that are present in Akey - LOG NORMALIZE
```{r}
ab1 = vector(mode="list", length = length(ab))
 for (i in 1:length(ab)){
   for (h in 1:length(names(ab[[i]]))){
     ab1[[i]][[h]] <- ab[[i]][h]
   }
 }
 
for (i in 1:length(ab)){
   for (h in 1:length(names(ab[[i]]))){
     G_list <- getBM(filters= "ensembl_gene_id", attributes= c("ensembl_gene_id", "hgnc_symbol"),values=rownames(ab1[[i]][[h]]),mart=ensembl)
     ab1[[i]][[h]]['gene_name'] <-  G_list$hgnc_symbol[match(rownames(ab1[[i]][[h]]), G_list$ensembl_gene_id)]
     ab1[[i]][[h]] <-ab1[[i]][[h]][order(ab1[[i]][[h]][[1]], decreasing = TRUE), ]
   }
}

#Intersecting q10 (high expression) with Akey
akey10 = vector(mode="list", length = length(ab)) #Creating empty list with 5 elements as ab
for (i in 1:length(ab)){ #(h in 1:length(names(ab[[i]]))) to generate same number of dataframes (in this case 16) as original in ab 
  for (h in 1:length(names(ab[[i]]))){
    akey10[[i]][[h]] <- ab1[[i]][[h]][ab1[[i]][[h]][[2]] %in% results$hgnc_symbol,] #in Akey
    akey10[[i]][[h]] <-  akey10[[i]][[h]][!(is.na(akey10[[i]][[h]][[2]]) | akey10[[i]][[h]][[2]]==""), ]
    akey10[[i]][[h]][1] <- log2(akey10[[i]][[h]][1]+1)
    #Cleaning
  }
}

#Reporting mean
mean1 = vector(mode="list", length = length(ab))
for (i in 1:length(ab)){
  for (h in 1:length(names(ab[[i]]))){
    mean1[[i]][[h]] <- median(akey10[[i]][[h]][[1]]) #mean
    mean1[[i]][[h]] <- as.data.frame(mean1[[i]][[h]])
    names(mean1[[i]][[h]]) <- paste(names(akey10[[i]][[h]][1]), sep='_')
  }
}

prenatal <- do.call("rbind", as.data.frame(mean1[[1]]))
colnames(prenatal) <- "prenatal"
infant <- do.call("rbind", as.data.frame(mean1[[2]]))
colnames(infant) <- "infant"
child <- do.call("rbind", as.data.frame(mean1[[3]]))
colnames(child) <- "child"
adolescent <- do.call("rbind", as.data.frame(mean1[[4]]))
colnames(adolescent) <- "adolescent"
adult <- do.call("rbind", as.data.frame(mean1[[5]]))
colnames(adult) <- "adult"

final_merge <- as.data.frame(cbind(prenatal, infant, child, adolescent, adult))
#write.xlsx(final_merge,row.names = TRUE, file = "~/desertsHomo/1.data/ABAData_AkeyPeyRac_log2/ABA_GenesAkey_log2.xlsx")

#Plot
final_merge <- tibble::rownames_to_column(final_merge, "Structure")
final_merge[[1]] <- sapply(strsplit(final_merge[[1]], "_"), "[", 1)

a<-ggparcoord(final_merge,
              columns = 2:6, groupColumn = 1, showPoints = TRUE, scale = "globalminmax",title="Genes in Deserts")+scale_color_viridis(discrete=TRUE)+theme(plot.title = element_text(size=10))+xlab("")+ylab("expression")
b <- ggparcoord(final_merge,
                columns = 2:6, groupColumn = 1, showPoints = TRUE, scale = "globalminmax",title="Striatum")+scale_color_manual(values = c( "#ABABAB", "#ABABAB", "#ABABAB", "#ABABAB","#ABABAB", "#ABABAB", "#ABABAB",  "#ABABAB", "#ABABAB", "#ABABAB","#ABABAB", "#ABABAB", "#ABABAB", "#FF0000", "#ABABAB", "#ABABAB"))+theme(plot.title = element_text(size=10),legend.position = "none")+xlab("")+ylab("expression")
c<-ggparcoord(final_merge,
              columns = 2:6, groupColumn = 1, showPoints = TRUE, scale = "globalminmax",title="Cerebellum")+scale_color_manual(values = c( "#ABABAB", "#ABABAB", "#0000FF", "#ABABAB","#ABABAB", "#ABABAB", "#ABABAB",  "#ABABAB", "#ABABAB", "#ABABAB","#ABABAB", "#ABABAB", "#ABABAB", "#ABABAB", "#ABABAB", "#ABABAB"))+theme(plot.title = element_text(size=10), legend.position = "none")+xlab("")+ylab("expression")
a1<-arrangeGrob(a, left=textGrob("A")) 
b1<-arrangeGrob(b, left =textGrob("B"))
c1<-arrangeGrob(c, left=textGrob("C"))
grid.arrange(a1, arrangeGrob(b1, c1), ncol = 2)

pl1abak <-grid.arrange(a1, b1, c1, ncol = 2, layout_matrix = rbind(c(1, 1, 2), c(1, 1, 3)))
##ggsave(file="~/desertsHomo/2.plots/ABAData_AkeyPey_log2_median/ABA_GenesAkey_log_median.pdf", pl1abak, width = 11.69, height = 8.27, units = "in")
```

#ABA Data - All genes that are present in both Akey and Pey LOG NORMALIZE
```{r}
#Global data
ab2 = vector(mode="list", length = length(ab))
 for (i in 1:length(ab)){
   for (h in 1:length(names(ab[[i]]))){
     ab2[[i]][[h]] <- ab[[i]][h]
   }
 }
 
for (i in 1:length(ab)){
   for (h in 1:length(names(ab[[i]]))){
     G_list <- getBM(filters= "ensembl_gene_id", attributes= c("ensembl_gene_id", "hgnc_symbol"),values=rownames(ab2[[i]][[h]]),mart=ensembl)
     ab2[[i]][[h]]['gene_name'] <-  G_list$hgnc_symbol[match(rownames(ab2[[i]][[h]]), G_list$ensembl_gene_id)]
     ab2[[i]][[h]] <-ab2[[i]][[h]][order(ab2[[i]][[h]][[1]], decreasing = TRUE), ]
   }
}
akeypey10 = vector(mode="list", length = length(ab)) #Creating empty list with 5 elements as ab
for (i in 1:length(ab)){ #(h in 1:length(names(ab[[i]]))) to generate same number of dataframes (in this case 16) as original in ab 
  for (h in 1:length(names(ab[[i]]))){
    akeypey10[[i]][[h]] <- ab2[[i]][[h]][ab2[[i]][[h]][[2]] %in% both$hgnc_symbol,] #in Akey
    akeypey10[[i]][[h]] <-  akeypey10[[i]][[h]][!(is.na(akeypey10[[i]][[h]][[2]]) | akeypey10[[i]][[h]][[2]]==""), ]
    akeypey10[[i]][[h]][1] <- log2(akeypey10[[i]][[h]][1]+1)
#Cleaning
  }
}

mean2 = vector(mode="list", length = length(ab))
for (i in 1:length(ab)){
  for (h in 1:length(names(ab[[i]]))){
    mean2[[i]][[h]] <- median(akeypey10[[i]][[h]][[1]]) #mean
    mean2[[i]][[h]] <- as.data.frame(mean2[[i]][[h]])
    names(mean2[[i]][[h]]) <- paste(names(akeypey10[[i]][[h]][1]), sep='_')
  }
}
prenatal <- do.call("rbind", as.data.frame(mean2[[1]]))
colnames(prenatal) <- "prenatal"
infant <- do.call("rbind", as.data.frame(mean2[[2]]))
colnames(infant) <- "infant"
child <- do.call("rbind", as.data.frame(mean2[[3]]))
colnames(child) <- "child"
adolescent <- do.call("rbind", as.data.frame(mean2[[4]]))
colnames(adolescent) <- "adolescent"
adult <- do.call("rbind", as.data.frame(mean2[[5]]))
colnames(adult) <- "adult"

final_mergelog <- as.data.frame(cbind(prenatal, infant, child, adolescent, adult))
#write.xlsx(final_mergelog,row.names = TRUE, file = "~/desertsHomo/1.data/ABAData_AkeyPeyRac_log2/ABA_GenesAkeyPey_log2.xlsx")

#For plot:
final_mergelog <- tibble::rownames_to_column(final_mergelog, "Structure")
final_mergelog[[1]] <- sapply(strsplit(final_mergelog[[1]], "_"), "[", 1)

a<-ggparcoord(final_mergelog,
    columns = 2:6, groupColumn = 1, showPoints = TRUE, scale = "globalminmax",title="Genes in Deserts and Pey")+scale_color_viridis(discrete=TRUE)+theme(plot.title = element_text(size=10))+xlab("")+ylab("expression")
b <- ggparcoord(final_mergelog,
    columns = 2:6, groupColumn = 1, showPoints = TRUE, scale = "globalminmax",title="Striatum (red) & MD (green)")+scale_color_manual(values = c( "#ABABAB", "#ABABAB", "#ABABAB", "#ABABAB","#ABABAB", "#ABABAB", "#ABABAB",  "#ABABAB", "#2ca25f", "#ABABAB","#ABABAB", "#ABABAB", "#ABABAB", "#FF0000", "#ABABAB", "#ABABAB"))+theme(plot.title = element_text(size=10),legend.position = "none")+xlab("")+ylab("expression")
c<-ggparcoord(final_mergelog,
    columns = 2:6, groupColumn = 1, showPoints = TRUE, scale = "globalminmax",title="Cerebellum (blue) and HIP (orange)")+scale_color_manual(values = c( "#ABABAB", "#ABABAB", "#0000FF", "#ABABAB","#fdae6b", "#ABABAB", "#ABABAB",  "#ABABAB", "#ABABAB", "#ABABAB","#ABABAB", "#ABABAB", "#ABABAB", "#ABABAB", "#ABABAB", "#ABABAB"))+theme(plot.title = element_text(size=10), legend.position = "none")+xlab("")+ylab("expression")

a1<-arrangeGrob(a, left=textGrob("A"))
b1<-arrangeGrob(b, left =textGrob("B"))
c1<-arrangeGrob(c, left=textGrob("C"))
grid.arrange(a1, arrangeGrob(b1, c1), ncol = 2)
grid.arrange(a1, b1, c1, ncol = 2, layout_matrix = rbind(c(1, 1, 2), c(1, 1, 3)))

d<-ggparcoord(final_mergelog,
    columns = 2:6, groupColumn = 1, showPoints = TRUE, scale = "globalminmax",title="Somato - Motor - Parietal - Aud Ctx")+scale_color_manual(values = c( "#00FF00", "#ABABAB", "#ABABAB", "#ABABAB","#ABABAB", "#00FF00", "#ABABAB",  "#00FF00", "#ABABAB", "#ABABAB","#ABABAB", "#00FF00", "#ABABAB", "#ABABAB", "#ABABAB", "#ABABAB"))+theme(plot.title = element_text(size=10), legend.position = "none")+xlab("")+ylab("expression")
d1<-arrangeGrob(d, left=textGrob("D"))

plakap <- grid.arrange(a1, b1, c1, ncol = 2, layout_matrix = rbind(c(1, 1, 2), c(1, 1, 3)))
##ggsave(file="~/desertsHomo/2.plots/ABAData_AkeyPey_log2_median/ABA_GenesAkeyPey_log_median.pdf", plakap, width = 11.69, height = 8.27, units = "in")
```

#ABA Data - All genes (raw data) LOG NORMALIZE
```{r}
#ab
aba_all = vector(mode="list", length = length(ab)) #Creating empty list with 5 elements as ab
for (i in 1:length(ab)){ #(h in 1:length(names(ab[[i]]))) to generate same number of dataframes (in this case 16) as original in ab 
  for (h in 1:length(names(ab[[i]]))){
    aba_all[[i]][[h]] <- q10[[i]][[h]][q10[[i]][[h]][[2]] %in% resP$hgnc_symbol,]
    aba_all[[i]][[h]] <-  aba_all[[i]][[h]][!(is.na(aba_all[[i]][[h]][[2]]) | aba_all[[i]][[h]][[2]]==""), ]
    aba_all[[i]][[h]][1] <- log2(aba_all[[i]][[h]][1])
#Cleaning
  }
}

mean0 = vector(mode="list", length = length(ab))
for (i in 1:length(ab)){
  for (h in 1:length(names(ab[[i]]))){
    mean0[[i]][[h]] <- mean(aba_all[[i]][[h]][[1]])
    mean0[[i]][[h]] <- as.data.frame(mean0[[i]][[h]])
    names(mean0[[i]][[h]]) <- paste(names(aba_all[[i]][[h]][1]), sep='_')
  }
}

prenatal <- do.call("rbind", as.data.frame(mean0[[1]]))
colnames(prenatal) <- "prenatal"
infant <- do.call("rbind", as.data.frame(mean0[[2]]))
colnames(infant) <- "infant"
child <- do.call("rbind", as.data.frame(mean0[[3]]))
colnames(child) <- "child"
adolescent <- do.call("rbind", as.data.frame(mean0[[4]]))
colnames(adolescent) <- "adolescent"
adult <- do.call("rbind", as.data.frame(mean0[[5]]))
colnames(adult) <- "adult"

final_merge0 <- as.data.frame(cbind(prenatal, infant, child, adolescent, adult))
#For plot
final_merge0 <- tibble::rownames_to_column(final_merge0, "Structure")
final_merge0[[1]] <- sapply(strsplit(final_merge0[[1]], "_"), "[", 1)

a<-ggparcoord(final_merge0,
    columns = 2:6, groupColumn = 1, showPoints = TRUE, scale = "globalminmax",title="Genes Raw Data")+scale_color_viridis(discrete=TRUE)+theme(plot.title = element_text(size=10))+xlab("")+ylab("mean  expression")
b <- ggparcoord(final_merge0,
    columns = 2:6, groupColumn = 1, showPoints = TRUE, scale = "globalminmax",title="Striatum")+scale_color_manual(values = c( "#ABABAB", "#ABABAB", "#ABABAB", "#ABABAB","#ABABAB", "#ABABAB", "#ABABAB",  "#ABABAB", "#ABABAB", "#ABABAB","#ABABAB", "#ABABAB", "#ABABAB", "#FF0000", "#ABABAB", "#ABABAB"))+theme(plot.title = element_text(size=10),legend.position = "none")+xlab("")+ylab("expression")
c<-ggparcoord(final_merge0,
    columns = 2:6, groupColumn = 1, showPoints = TRUE, scale = "globalminmax",title="Cerebellum")+scale_color_manual(values = c( "#ABABAB", "#ABABAB", "#0000FF", "#ABABAB","#ABABAB", "#ABABAB", "#ABABAB",  "#ABABAB", "#ABABAB", "#ABABAB","#ABABAB", "#ABABAB", "#ABABAB", "#ABABAB", "#ABABAB", "#ABABAB"))+theme(plot.title = element_text(size=10), legend.position = "none")+xlab("")+ylab("expression")
a1<-arrangeGrob(a, left=textGrob("A"))
b1<-arrangeGrob(b, left =textGrob("B"))
c1<-arrangeGrob(c, left=textGrob("C"))
grid.arrange(a1, arrangeGrob(b1, c1), ncol = 2)


pl3 <-grid.arrange(a1, b1, c1, ncol = 2, layout_matrix = rbind(c(1, 1, 2), c(1, 1, 3)))
###ggsave(file="~/desertsHomo/2.plots/ABAData_AkeyPeyRac_log2/ABA_GenesAll_log.pdf", pl3, width = 11.69, height = 8.27, units = "in")
```

#Raw Sestan
```{r}
mRNAseqData=read.table("~/tmp_psychENCODE/mRNA-seq_hg38.gencode21.wholeGene.geneComposite.STAR.nochrM.gene.RPKM.normalized.CQNCombat.txt",sep="\t",header=TRUE)
modsb1= mRNAseqData %>% 
  separate(Geneid,c("EnsemblID","Genename"),extra="merge")
modsb1$EnsemblID<-NULL
rawSestan <- modsb1
rawSestan=t(rawSestan)
#As dataframe
rawSestan=as.data.frame(rawSestan)
colnames(rawSestan) <- as.matrix(unlist(rawSestan[1,]))
rawSestan <- rawSestan[-1, ]
rawSestan <- cbind(info = rownames(rawSestan), rawSestan)
rownames(rawSestan) <- 1:nrow(rawSestan)
#duplicated columns - Remove duplicates if needed
#colnames(rawSestan)[duplicated(colnames(rawSestan))] #0
rawSestan <- rawSestan[, !duplicated(colnames(rawSestan))]
rawSestan=rawSestan %>% 
  separate(info, c("Braincode","Regioncode"))
lograwsestan1 <- rawSestan
cols.num <- colnames(lograwsestan1[3:ncol(lograwsestan1)])
lograwsestan1[,cols.num] <- lapply(lograwsestan1[cols.num],as.character)
lograwsestan1[,cols.num] <- lapply(lograwsestan1[cols.num],as.numeric)
lograwsestan1[3:ncol(lograwsestan1)] <- log2(lograwsestan1[3:ncol(lograwsestan1)]+1)
metadatamRNAseq=read_xlsx("~/tmp_psychENCODE/mRNA-seq_QC.xlsx",skip = 3)
modMetadatamRNAseq=na.omit(metadatamRNAseq)
modMetadatamRNAseq=modMetadatamRNAseq %>% select(1:3)
modMetadatamRNAseq=as.data.frame(modMetadatamRNAseq)
finalrawSestan=merge(modMetadatamRNAseq,lograwsestan1,by=c("Braincode", "Regioncode"))
finalraw2 <- finalrawSestan
rownames(finalraw2)<- do.call(paste,c(finalraw2[c("Braincode","Regioncode", "Window")],sep="_"))
finalraw2[(1:3)] <- NULL
finalraw2<-finalraw2 %>% select(which(colMedians(as.matrix(finalraw2))>2))
library(tibble)
finalraw2 <- rownames_to_column(finalraw2) 
finalraw2 <- finalraw2 %>% 
  separate(rowname, c("Braincode","Regioncode", "Window")) #Input to PC
finalraw2$Braincode <- NULL
finalraw2$Label<- paste(finalraw2$Regioncode, finalraw2$Window, sep="_")
finalraw2$Regioncode <- NULL
finalraw2$Window <- NULL
finalraw2 <- as_tibble(finalraw2)
finalraw2_w_mean <- finalraw2 %>%
  group_by(Label) %>%
  dplyr::summarise_all(median, na.rm=TRUE)
dfrawSestan2 <- data.frame(Structure=finalraw2_w_mean$Label, Means=rowMeans(finalraw2_w_mean[,-1]))
dfrawSestan2 <- dfrawSestan2 %>% 
  separate(Structure, c("Structure","Window"))
df_raw2 <- pivot_wider(dfrawSestan2, names_from = Window, values_from = Means)
df_raw2[,10] <- NULL
df_raw2 <- df_raw2[complete.cases(df_raw2), ]
colnames(df_raw2) <- c("Structure", "Fetal_1", "Fetal_2", "Fetal_3", "Birth/Ifan", "Infan/Child", "Child", "Adolescence", "Adult")
#PLOT
levels(colnames(df_raw2)) <- c("Structure", "Fetal_1", "Fetal_2", "Fetal_3", "Birth/Inf", "Inf/Child", "Child", "Adolescence", "Adult")
##write.csv(df_raw2, file="median_filtered_rawSestan.csv", row.names = FALSE)
a<- ggparcoord(df_raw2,
               columns = 2:9, groupColumn = 1, showPoints = TRUE, scale = "globalminmax",title="Genes in Deserts- VFC (green) & AMY (black)")+scale_color_manual(values = c( "#ABABAB", "#000000", "#ABABAB", "#ABABAB","#ABABAB", "#ABABAB", "#ABABAB",  "#ABABAB", "#ABABAB", "#ABABAB","#ABABAB", "#ABABAB", "#ABABAB", "#ABABAB", "#ABABAB", "#238b45"))+theme(plot.title = element_text(size=10),legend.position = "none")+xlab("")+ylab("expression")
b <- ggparcoord(df_raw2,
                columns = 2:9, groupColumn = 1, showPoints = TRUE, scale = "globalminmax",title="Striatum")+scale_color_manual(values = c( "#ABABAB", "#ABABAB", "#ABABAB", "#ABABAB","#ABABAB", "#ABABAB", "#ABABAB",  "#ABABAB", "#ABABAB", "#ABABAB","#ABABAB", "#ABABAB", "#ABABAB", "#FF0000", "#ABABAB", "#ABABAB"))+theme(plot.title = element_text(size=10),legend.position = "none")+xlab("")+ylab("expression")
c<-ggparcoord(df_raw2,
              columns = 2:9, groupColumn = 1, showPoints = TRUE, scale = "globalminmax",title="Cerebellum")+scale_color_manual(values = c( "#ABABAB", "#ABABAB", "#0000FF", "#ABABAB","#ABABAB", "#ABABAB", "#ABABAB",  "#ABABAB", "#ABABAB", "#ABABAB","#ABABAB", "#ABABAB", "#ABABAB", "#ABABAB", "#ABABAB", "#ABABAB"))+theme(plot.title = element_text(size=10), legend.position = "none")+xlab("")+ylab("expression")
a1<-arrangeGrob(a, left=textGrob("A"))
b1<-arrangeGrob(b, left =textGrob("B"))
c1<-arrangeGrob(c, left=textGrob("C"))
dfraw2_pl <-grid.arrange(a1, b1, c1, ncol = 2, layout_matrix = rbind(c(1, 1, 2), c(1, 1, 3)))
##ggsave(file="filtered_rawSestan_log2_median.pdf", dfraw2_pl, width = 11.69, height = 8.27, units = "in")
```

#Akey in Sestan data
```{r}
mRNAseqData=read.table("~/tmp_psychENCODE/mRNA-seq_hg38.gencode21.wholeGene.geneComposite.STAR.nochrM.gene.RPKM.normalized.CQNCombat.txt",sep="\t",header=TRUE)
modsb1= mRNAseqData %>% 
  separate(Geneid,c("EnsemblID","Genename"),extra="merge")
modsb1$EnsemblID<-NULL
#Filtering for Akey alone
akeySestan1 <- modsb1 %>% filter(modsb1$Genename %in% results$hgnc_symbol)
akeySestan1=t(akeySestan1)

#As dataframe
akeySestan1=as.data.frame(akeySestan1)

colnames(akeySestan1) <- as.matrix(unlist(akeySestan1[1,]))
akeySestan1 <- akeySestan1[-1, ]

akeySestan1 <- cbind(info = rownames(akeySestan1), akeySestan1)
rownames(akeySestan1) <- 1:nrow(akeySestan1)

#duplicated columns - issue in raw data. Here: 
colnames(akeySestan1)[duplicated(colnames(akeySestan1))] #1
akeySestan1 <- akeySestan1[, !duplicated(colnames(akeySestan1))]

akeySestan1=akeySestan1 %>% 
  separate(info, c("Braincode","Regioncode"))

#Normality
#normcheck <- akeypeySestan %>% pivot_longer(!("Braincode"|"Regioncode"), names_to = "Genes", values_to = "RPKM")
#normcheck$RPKM <- as.numeric(normcheck$RPKM)
#ggqqplot(normcheck, "RPKM", facet.by = "Regioncode")

#Transformation
logakeySestan1 <- akeySestan1

cols.num <- colnames(logakeySestan1[3:ncol(logakeySestan1)])

logakeySestan1[,cols.num] <- lapply(logakeySestan1[cols.num],as.character)
logakeySestan1[,cols.num] <- lapply(logakeySestan1[cols.num],as.numeric)

logakeySestan1[3:ncol(logakeySestan1)] <- log2(logakeySestan1[3:ncol(logakeySestan1)]+1)

#Normality after transformation
#lognormcheck <- logakeypeySestan %>% pivot_longer(!("Braincode"|"Regioncode"), names_to = "Genes", values_to = "RPKM")
#lognormcheck$RPKM <- as.numeric(lognormcheck$RPKM)
#ggqqplot(lognormcheck, "RPKM", facet.by = "Regioncode")


#Brining the metadata of the database
library(readxl)
metadatamRNAseq=read_xlsx("~/tmp_psychENCODE/mRNA-seq_QC.xlsx",skip = 3)
modMetadatamRNAseq=na.omit(metadatamRNAseq)
modMetadatamRNAseq=modMetadatamRNAseq %>% select(1:3)
modMetadatamRNAseq=as.data.frame(modMetadatamRNAseq)

#With log
finalakeySestan1=merge(modMetadatamRNAseq,logakeySestan1,by=c("Braincode", "Regioncode")) #input
 
# #write.csv(finalakeySestan1,"logakey.csv")

#Without log:
#finalakeySestan1=merge(modMetadatamRNAseq,akeypeySestan,by=c("Braincode", "Regioncode"))

finalakeySestan1$Braincode <- NULL
finalakeySestan1$Label<- paste(finalakeySestan1$Regioncode, finalakeySestan1$Window, sep="_")
finalakeySestan1$Regioncode <- NULL
finalakeySestan1$Window <- NULL

finalakeySestan1 <- as_tibble(finalakeySestan1)

finalakeySestan1_w_mean <- finalakeySestan1 %>%
    group_by(Label) %>%
    dplyr::summarise_all(median, na.rm=TRUE)

#For plotting
dfakeySestan1 <- data.frame(Structure=finalakeySestan1_w_mean$Label, Means=rowMeans(finalakeySestan1_w_mean[,-1]))

dfakeySestan1 <- dfakeySestan1 %>% 
  separate(Structure, c("Structure","Window"))
preli1 <- pivot_wider(dfakeySestan1, names_from = Window, values_from = Means)
preli1[,10] <- NULL
preli1 <- preli1[complete.cases(preli1), ]
colnames(preli1) <- c("Structure", "Fetal_1", "Fetal_2", "Fetal_3", "Birth/Ifan", "Infan/Child", "Child", "Adolescence", "Adult")
#PLOT
levels(colnames(preli1)) <- c("Structure", "Fetal_1", "Fetal_2", "Fetal_3", "Birth/Inf", "Inf/Child", "Child", "Adolescence", "Adult")

#write.csv(preli1, file="~/desertsHomo/1.data/median_tables/median_Akey_Sestan.csv", row.names = FALSE)

a<- ggparcoord(preli1,
              columns = 2:9, groupColumn = 1, showPoints = TRUE, scale = "globalminmax",title="Genes in Deserts- VFC (green) & AMY (black)")+scale_color_manual(values = c( "#ABABAB", "#000000", "#ABABAB", "#ABABAB","#ABABAB", "#ABABAB", "#ABABAB",  "#ABABAB", "#ABABAB", "#ABABAB","#ABABAB", "#ABABAB", "#ABABAB", "#ABABAB", "#ABABAB", "#238b45"))+theme(plot.title = element_text(size=10),legend.position = "none")+xlab("")+ylab("expression")
b <- ggparcoord(preli1,
                 columns = 2:9, groupColumn = 1, showPoints = TRUE, scale = "globalminmax",title="Striatum")+scale_color_manual(values = c( "#ABABAB", "#ABABAB", "#ABABAB", "#ABABAB","#ABABAB", "#ABABAB", "#ABABAB",  "#ABABAB", "#ABABAB", "#ABABAB","#ABABAB", "#ABABAB", "#ABABAB", "#FF0000", "#ABABAB", "#ABABAB"))+theme(plot.title = element_text(size=10),legend.position = "none")+xlab("")+ylab("expression")
c<-ggparcoord(preli1,
               columns = 2:9, groupColumn = 1, showPoints = TRUE, scale = "globalminmax",title="Cerebellum")+scale_color_manual(values = c( "#ABABAB", "#ABABAB", "#0000FF", "#ABABAB","#ABABAB", "#ABABAB", "#ABABAB",  "#ABABAB", "#ABABAB", "#ABABAB","#ABABAB", "#ABABAB", "#ABABAB", "#ABABAB", "#ABABAB", "#ABABAB"))+theme(plot.title = element_text(size=10), legend.position = "none")+xlab("")+ylab("expression")
a1<-arrangeGrob(a, left=textGrob("A"))
b1<-arrangeGrob(b, left =textGrob("B"))
c1<-arrangeGrob(c, left=textGrob("C"))
grid.arrange(a1, arrangeGrob(b1, c1), ncol = 2)
# 
plak_sestan <-grid.arrange(a1, b1, c1, ncol = 2, layout_matrix = rbind(c(1, 1, 2), c(1, 1, 3)))
#ggsave(file="~/desertsHomo/2.plots/Sestan_AkeyPey_log2_median/Sestan_GenesAkey_log2_median.pdf", plak_sestan, width = 11.69, height = 8.27, units = "in")

```

#AkeyPey in Sestan data
```{r}
mRNAseqData=read.table("~/Desktop/gcloud_2021_06/rstudio_2021_06/tmp_psychENCODE/mRNA-seq_hg38.gencode21.wholeGene.geneComposite.STAR.nochrM.gene.RPKM.normalized.CQNCombat.txt",sep="\t",header=TRUE)
modsb1= mRNAseqData %>% 
  separate(Geneid,c("EnsemblID","Genename"),extra="merge")
modsb1$EnsemblID<-NULL
#Filtering for Akey alone
akeypeySestan <- modsb1 %>% filter(modsb1$Genename %in% both$hgnc_symbol)
akeypeySestan=t(akeypeySestan)



#As dataframe
akeypeySestan=as.data.frame(akeypeySestan)

colnames(akeypeySestan) <- as.matrix(unlist(akeypeySestan[1,]))
akeypeySestan <- akeypeySestan[-1, ]

akeypeySestan <- cbind(info = rownames(akeypeySestan), akeypeySestan)
rownames(akeypeySestan) <- 1:nrow(akeypeySestan)

#duplicated columns - issue in raw data. Here: 
colnames(akeypeySestan)[duplicated(colnames(akeypeySestan))] #0
#akeypeySestan <- akeypeySestan[, !duplicated(colnames(akeypeySestan))]

akeypeySestan=akeypeySestan %>% 
  separate(info, c("Braincode","Regioncode"))

# #Normality
# normcheck <- akeypeySestan %>% pivot_longer(!("Braincode"|"Regioncode"), names_to = "Genes", values_to = "RPKM")
# normcheck$RPKM <- as.character(normcheck$RPKM)
# normcheck$RPKM <- as.numeric(normcheck$RPKM)
# 
# norm_1=merge(modMetadatamRNAseq,normcheck,by=c("Braincode", "Regioncode"))
# 
# norm_1 <- as_tibble(norm_1)
# 
# ggqqplot(norm_1, "RPKM", facet.by = "Window")
# 
# #Transformation
logakeypeysestan1 <- akeypeySestan
# 
cols.num <- colnames(logakeypeysestan1[3:ncol(logakeypeysestan1)])
# 
logakeypeysestan1[,cols.num] <- lapply(logakeypeysestan1[cols.num],as.character)
logakeypeysestan1[,cols.num] <- lapply(logakeypeysestan1[cols.num],as.numeric)
# 
logakeypeysestan1[3:ncol(logakeypeysestan1)] <- log2(logakeypeysestan1[3:ncol(logakeypeysestan1)]+1)
#ta <- logakeypeysestan1 %>% filter(logakeypeysestan1[3:ncol(logakeypeysestan1)]>1)

# 
# #Normality after transformation
# lognormcheck <- logakeypeysestan1 %>% pivot_longer(!("Braincode"|"Regioncode"), names_to = "Genes", values_to = "RPKM")
# lognormcheck$RPKM <- as.character(lognormcheck$RPKM)
# lognormcheck$RPKM <- as.numeric(lognormcheck$RPKM)
# lognorm_1=merge(modMetadatamRNAseq,lognormcheck,by=c("Braincode", "Regioncode"))
# # 
# lognorm_1 <- as_tibble(lognorm_1)
# # 
# ggqqplot(lognorm_1, "RPKM", facet.by = "Window")
# 

#Brining the metadata of the database
library(readxl)
metadatamRNAseq=read_xlsx("~/Desktop/gcloud_2021_06/rstudio_2021_06/tmp_psychENCODE/mRNA-seq_QC.xlsx",skip = 3)


modMetadatamRNAseq=na.omit(metadatamRNAseq)
modMetadatamRNAseq=modMetadatamRNAseq %>% dplyr::select(1:3)
modMetadatamRNAseq=as.data.frame(modMetadatamRNAseq)

#With log
finalakeypeySestan=merge(modMetadatamRNAseq,logakeypeysestan1,by=c("Braincode", "Regioncode"))
# #write.csv(finalakeypeySestan,"logakeypey.csv")
#Without log:
#finalakeypeySestan=merge(modMetadatamRNAseq,akeypeySestan,by=c("Braincode", "Regioncode"))

finalakeypeySestan$Braincode <- NULL
finalakeypeySestan$Label<- paste(finalakeypeySestan$Regioncode, finalakeypeySestan$Window, sep="_")
finalakeypeySestan$Regioncode <- NULL
finalakeypeySestan$Window <- NULL

finalakeypeySestan <- as_tibble(finalakeypeySestan)

#For plotting_0
g0 <- finalakeypeySestan %>%
  pivot_longer(!Label, names_to = "Gene", values_to = "expr")
g0 <- g0 %>% 
  separate(Label, c("Structure","Window"))

ggplot(g0, aes(x=Window,y=expr, colour=Structure)) + geom_line(aes(colour = factor(Structure)))+
  geom_point(aes(group = factor(Structure)))+
  stat_smooth(method="loess", span=0.1, se=TRUE, aes(fill=Structure, alpha=0.3) +
  theme_bw())
#
finalakeypeySestan_w_mean <- finalakeypeySestan %>%
    group_by(Label) %>%
    dplyr::summarise_all(median, na.rm=TRUE)


#For plotting_1
dfakeypeySestan <- data.frame(Structure=finalakeypeySestan_w_mean$Label, Means=rowMeans(finalakeypeySestan_w_mean[,-1]))

dfakeypeySestan <- dfakeypeySestan %>% 
  separate(Structure, c("Structure","Window"))
preli2 <- pivot_wider(dfakeypeySestan, names_from = Window, values_from = Means)
preli2[,10] <- NULL
preli2 <- preli2[complete.cases(preli2), ]
colnames(preli2) <- c("Structure", "Fetal_1", "Fetal_2", "Fetal_3", "Birth/Ifan", "Infan/Child", "Child", "Adolescence", "Adult")
#PLOT
levels(colnames(preli2)) <- c("Structure", "Fetal_1", "Fetal_2", "Fetal_3", "Birth/Inf", "Inf/Child", "Child", "Adolescence", "Adult")

#write.csv(preli2, file="~/desertsHomo/1.data/median_tables/median_AkeyPey_Sestan.csv", row.names = FALSE)

a<- ggparcoord(preli2,
              columns = 2:9, groupColumn = 1, showPoints = TRUE, scale = "globalminmax",title="Genes in Deserts and Pos Sel- VFC (green) & AMY (black)")+scale_color_manual(values = c( "#ABABAB", "#000000", "#ABABAB", "#ABABAB","#ABABAB", "#ABABAB", "#ABABAB",  "#ABABAB", "#ABABAB", "#ABABAB","#ABABAB", "#ABABAB", "#ABABAB", "#ABABAB", "#ABABAB", "#238b45"))+theme(plot.title = element_text(size=10),legend.position = "none")+xlab("")+ylab("expression")
b <- ggparcoord(preli2,
                 columns = 2:9, groupColumn = 1, showPoints = TRUE, scale = "globalminmax",title="Striatum")+scale_color_manual(values = c( "#ABABAB", "#ABABAB", "#ABABAB", "#ABABAB","#ABABAB", "#ABABAB", "#ABABAB",  "#ABABAB", "#ABABAB", "#ABABAB","#ABABAB", "#ABABAB", "#ABABAB", "#FF0000", "#ABABAB", "#ABABAB"))+theme(plot.title = element_text(size=10),legend.position = "none")+xlab("")+ylab("expression")
c<-ggparcoord(preli2,
               columns = 2:9, groupColumn = 1, showPoints = TRUE, scale = "globalminmax",title="Cerebellum")+scale_color_manual(values = c( "#ABABAB", "#ABABAB", "#0000FF", "#ABABAB","#ABABAB", "#ABABAB", "#ABABAB",  "#ABABAB", "#ABABAB", "#ABABAB","#ABABAB", "#ABABAB", "#ABABAB", "#ABABAB", "#ABABAB", "#ABABAB"))+theme(plot.title = element_text(size=10), legend.position = "none")+xlab("")+ylab("expression")
a1<-arrangeGrob(a, left=textGrob("A"))
b1<-arrangeGrob(b, left =textGrob("B"))
c1<-arrangeGrob(c, left=textGrob("C"))
grid.arrange(a1, arrangeGrob(b1, c1), ncol = 2)
 
plakpey_sestan <-grid.arrange(a1, b1, c1, ncol = 2, layout_matrix = rbind(c(1, 1, 2), c(1, 1, 3)))

##ggsave(file="~/desertsHomo/2.plots/Sestan_AkeyPey_log2_median/Sestan_GenesAkeyPey_log2_median.pdf", plakpey_sestan, width = 11.69, height = 8.27, units = "in")

```

# PeyNotAkey in Sestan data
```{r}
mRNAseqData=read.table("~/tmp_psychENCODE/mRNA-seq_hg38.gencode21.wholeGene.geneComposite.STAR.nochrM.gene.RPKM.normalized.CQNCombat.txt",sep="\t",header=TRUE)
modsb1= mRNAseqData %>% 
  separate(Geneid,c("EnsemblID","Genename"),extra="merge")
modsb1$EnsemblID<-NULL
#Filtering for Akey alone
peyNotAkeySestan <- modsb1 %>% filter(modsb1$Genename %in% both$hgnc_symbol)
peyNotAkeySestan=t(peyNotAkeySestan)



#As dataframe
peyNotAkeySestan=as.data.frame(peyNotAkeySestan)

colnames(peyNotAkeySestan) <- as.matrix(unlist(peyNotAkeySestan[1,]))
peyNotAkeySestan <- peyNotAkeySestan[-1, ]

peyNotAkeySestan <- cbind(info = rownames(peyNotAkeySestan), peyNotAkeySestan)
rownames(peyNotAkeySestan) <- 1:nrow(peyNotAkeySestan)

#duplicated columns - issue in raw data. Here: 
colnames(peyNotAkeySestan)[duplicated(colnames(peyNotAkeySestan))] #0
#peyNotAkeySestan <- peyNotAkeySestan[, !duplicated(colnames(peyNotAkeySestan))]

peyNotAkeySestan=peyNotAkeySestan %>% 
  separate(info, c("Braincode","Regioncode"))

# #Normality
# normcheck <- peyNotAkeySestan %>% pivot_longer(!("Braincode"|"Regioncode"), names_to = "Genes", values_to = "RPKM")
# normcheck$RPKM <- as.character(normcheck$RPKM)
# normcheck$RPKM <- as.numeric(normcheck$RPKM)
# 
# norm_1=merge(modMetadatamRNAseq,normcheck,by=c("Braincode", "Regioncode"))
# 
# norm_1 <- as_tibble(norm_1)
# 
# ggqqplot(norm_1, "RPKM", facet.by = "Window")
# 
# #Transformation
logpeyNotAkeysestan1 <- peyNotAkeySestan
# 
cols.num <- colnames(logpeyNotAkeysestan1[3:ncol(logpeyNotAkeysestan1)])
# 
logpeyNotAkeysestan1[,cols.num] <- lapply(logpeyNotAkeysestan1[cols.num],as.character)
logpeyNotAkeysestan1[,cols.num] <- lapply(logpeyNotAkeysestan1[cols.num],as.numeric)
# 
logpeyNotAkeysestan1[3:ncol(logpeyNotAkeysestan1)] <- log2(logpeyNotAkeysestan1[3:ncol(logpeyNotAkeysestan1)]+1)
#ta <- logpeyNotAkeysestan1 %>% filter(logpeyNotAkeysestan1[3:ncol(logpeyNotAkeysestan1)]>1)

# 
# #Normality after transformation
# lognormcheck <- logpeyNotAkeysestan1 %>% pivot_longer(!("Braincode"|"Regioncode"), names_to = "Genes", values_to = "RPKM")
# lognormcheck$RPKM <- as.character(lognormcheck$RPKM)
# lognormcheck$RPKM <- as.numeric(lognormcheck$RPKM)
# lognorm_1=merge(modMetadatamRNAseq,lognormcheck,by=c("Braincode", "Regioncode"))
# # 
# lognorm_1 <- as_tibble(lognorm_1)
# # 
# ggqqplot(lognorm_1, "RPKM", facet.by = "Window")
# 

#Brining the metadata of the database
library(readxl)
metadatamRNAseq=read_xlsx("~/tmp_psychENCODE/mRNA-seq_QC.xlsx",skip = 3)


modMetadatamRNAseq=na.omit(metadatamRNAseq)
modMetadatamRNAseq=modMetadatamRNAseq %>% select(1:3)
modMetadatamRNAseq=as.data.frame(modMetadatamRNAseq)

#With log
finalpeyNotAkeySestan=merge(modMetadatamRNAseq,logpeyNotAkeysestan1,by=c("Braincode", "Regioncode"))
# #write.csv(finalpeyNotAkeySestan,"logpeyNotAkey.csv")
#Without log:
#finalpeyNotAkeySestan=merge(modMetadatamRNAseq,peyNotAkeySestan,by=c("Braincode", "Regioncode"))

finalpeyNotAkeySestan$Braincode <- NULL
finalpeyNotAkeySestan$Label<- paste(finalpeyNotAkeySestan$Regioncode, finalpeyNotAkeySestan$Window, sep="_")
finalpeyNotAkeySestan$Regioncode <- NULL
finalpeyNotAkeySestan$Window <- NULL

finalpeyNotAkeySestan <- as_tibble(finalpeyNotAkeySestan)

finalpeyNotAkeySestan_w_mean <- finalpeyNotAkeySestan %>%
  group_by(Label) %>%
  dplyr::summarise_all(median, na.rm=TRUE)


#For plotting
dfpeyNotAkeySestan <- data.frame(Structure=finalpeyNotAkeySestan_w_mean$Label, Means=rowMeans(finalpeyNotAkeySestan_w_mean[,-1]))

dfpeyNotAkeySestan <- dfpeyNotAkeySestan %>% 
  separate(Structure, c("Structure","Window"))
preli2 <- pivot_wider(dfpeyNotAkeySestan, names_from = Window, values_from = Means)
preli2[,10] <- NULL
preli2 <- preli2[complete.cases(preli2), ]
colnames(preli2) <- c("Structure", "Fetal_1", "Fetal_2", "Fetal_3", "Birth/Ifan", "Infan/Child", "Child", "Adolescence", "Adult")
#PLOT
levels(colnames(preli2)) <- c("Structure", "Fetal_1", "Fetal_2", "Fetal_3", "Birth/Inf", "Inf/Child", "Child", "Adolescence", "Adult")

#write.csv(preli2, file="~/desertsHomo/1.data/median_tables/median_peyNotAkey_Sestan.csv", row.names = FALSE)

a<- ggparcoord(preli2,
               columns = 2:9, groupColumn = 1, showPoints = TRUE, scale = "globalminmax",title="Genes in Deserts and Pos Sel- VFC (green) & AMY (black)")+scale_color_manual(values = c( "#ABABAB", "#000000", "#ABABAB", "#ABABAB","#ABABAB", "#ABABAB", "#ABABAB",  "#ABABAB", "#ABABAB", "#ABABAB","#ABABAB", "#ABABAB", "#ABABAB", "#ABABAB", "#ABABAB", "#238b45"))+theme(plot.title = element_text(size=10),legend.position = "none")+xlab("")+ylab("expression")
b <- ggparcoord(preli2,
                columns = 2:9, groupColumn = 1, showPoints = TRUE, scale = "globalminmax",title="Striatum")+scale_color_manual(values = c( "#ABABAB", "#ABABAB", "#ABABAB", "#ABABAB","#ABABAB", "#ABABAB", "#ABABAB",  "#ABABAB", "#ABABAB", "#ABABAB","#ABABAB", "#ABABAB", "#ABABAB", "#FF0000", "#ABABAB", "#ABABAB"))+theme(plot.title = element_text(size=10),legend.position = "none")+xlab("")+ylab("expression")
c<-ggparcoord(preli2,
              columns = 2:9, groupColumn = 1, showPoints = TRUE, scale = "globalminmax",title="Cerebellum")+scale_color_manual(values = c( "#ABABAB", "#ABABAB", "#0000FF", "#ABABAB","#ABABAB", "#ABABAB", "#ABABAB",  "#ABABAB", "#ABABAB", "#ABABAB","#ABABAB", "#ABABAB", "#ABABAB", "#ABABAB", "#ABABAB", "#ABABAB"))+theme(plot.title = element_text(size=10), legend.position = "none")+xlab("")+ylab("expression")
a1<-arrangeGrob(a, left=textGrob("A"))
b1<-arrangeGrob(b, left =textGrob("B"))
c1<-arrangeGrob(c, left=textGrob("C"))
grid.arrange(a1, arrangeGrob(b1, c1), ncol = 2)

plpeynotAk_sestan <-grid.arrange(a1, b1, c1, ncol = 2, layout_matrix = rbind(c(1, 1, 2), c(1, 1, 3)))

##ggsave(file="~/desertsHomo/2.plots/Sestan_peyNotAkey_log2_median/Sestan_GenespeyNotAkey_log2_median.pdf", plpeynotAk_sestan, width = 11.69, height = 8.27, units = "in")
```

#Trajectory plots1 - ABA
```{r}
final_merge0 #all
final_merge #akey
final_mergelog #akeypey

final_merge0$dataset <- c("raw")
final_merge$dataset <- c("akey")
final_mergelog$dataset <- c("akeypey")

tot_pl <-rbind(final_merge, final_merge0, final_mergelog)
tot_pl <- as_tibble(tot_pl)
tot_pl <- tot_pl %>% mutate(dataset=as.character(dataset))

levels(tot_pl$dataset) <-  c("raw", "akeypey", "akey")

n <-ggparcoord(tot_pl,
columns = 2:6, groupColumn = 1, showPoints = TRUE, scale = "globalminmax",title="Structures ABA")+theme(plot.title = element_text(size=10), legend.position = "none")+xlab("")+ylab("expression")



n <-ggparcoord(tot_pl,
columns = 2:6, groupColumn = 1, showPoints = TRUE, scale = "globalminmax",title="Structures ABA", mapping=aes(color=as.factor(dataset)))+theme(plot.title = element_text(size=10))+xlab("")+ylab("expression")+labs(color="Dataset")


n + facet_wrap(~Structure)

an <- n + facet_wrap(~Structure) 
##ggsave(file="~/desertsHomo/2.plots/ABAData_AkeyPeyRac_log2/ABA_temporal_Structures.pdf", an, width = 11.69, height = 8.27, units = "in")
```

#FIGURE #2
```{r}
traj_fig <- tot_pl %>% filter(Structure %in% "DFC" | Structure %in% "M1C" | Structure %in% "S1C" | Structure %in% "CBC") 

figtop1 <- ggparcoord(traj_fig,
           columns = 2:9, groupColumn = 1, showPoints = TRUE, scale = "globalminmax", mapping=aes(color=as.factor(dataset)), title="A")+theme(plot.title = element_text(size=10), axis.text.x = element_text(angle = 45,  hjust = 1))+xlab("")+ylab("expression")+labs(color="Dataset")+ facet_wrap(~Structure, ncol = 4)+scale_color_discrete(name="Dataset",labels=unique(tot_pl$dataset))

#Akey
vpreli1 <- preli1
vpreli1$Structure <- str_replace(vpreli1$Structure, ("A1C|DFC|IPC|ITC|M1C|MFC|OFC|S1C|STC|V1C|VFC"), "NCX")

test <- vpreli1 %>%
    group_by(Structure) %>%
    dplyr::summarise_all(mean, na.rm=TRUE)
figtop2<-ggparcoord(test,
           columns = 2:9, groupColumn = 1, showPoints = TRUE, scale = "globalminmax",title="B")+theme(plot.title = element_text(size=10),legend.position = "none", axis.text.x = element_text(angle = 45,  hjust = 1))+xlab("")+ylab("expression")+geom_line(size=1.7, alpha=0.7)

#AKeyPey
vpreli2 <- preli2
vpreli2$Structure <- str_replace(vpreli2$Structure, ("A1C|DFC|IPC|ITC|M1C|MFC|OFC|S1C|STC|V1C|VFC"), "NCX")

test2 <- vpreli2 %>%
    group_by(Structure) %>%
    dplyr::summarise_all(mean, na.rm=TRUE)
figtop3<-ggparcoord(test2,
           columns = 2:9, groupColumn = 1, showPoints = TRUE, scale = "globalminmax",title="C")+theme(plot.title = element_text(size=10),legend.position = "right", axis.text.x = element_text(angle = 45,  hjust = 1))+xlab("")+ylab("expression")+geom_line(size=1.7, alpha=0.7)

lay <- rbind(c(1,1,1,1,1,1),
             c(2,2,2,3,3,3), 
             c(2,2,2,3,3,3))

grid.arrange(figtop1, figtop2, figtop3, layout_matrix = lay)
fig2_main <- grid.arrange(figtop1, figtop2, figtop3, layout_matrix = lay)
ggsave(file="~/desertsHomo/2.plots/Sestan_AkeyPey_log2_median/fig_combined.pdf", fig2_main, width = 11.69, height = 8.27, units = "in")
```

#Fig3 - Boxplots Akey + Selected structures
```{r}
figtop3<-ggparcoord(test,
           columns = 2:9, groupColumn = 1, showPoints = TRUE, scale = "globalminmax",title="C")+theme(plot.title = element_text(size=10),legend.position = "right", axis.text.x = element_text(angle = 45,  hjust = 1))+xlab("")+ylab("expression")+geom_line(size=1.7, alpha=0.7)
leg1 <- get_legend(figtop3)
leg1 <- as_ggplot(leg1)#legend as figure

str_ak <-ggparcoord(test,
           columns = 2:9, groupColumn = 1, showPoints = TRUE, scale = "globalminmax",title="Striatum")+theme(plot.title = element_text(size=10),legend.position = "none", axis.text.x = element_text(angle = 45,  hjust = 1))+xlab("")+ylab("expression")+geom_line(size=1.7, alpha=0.7)+scale_color_manual(values = c( "#ABABAB", "#ABABAB", "#ABABAB", "#ABABAB", "#ABABAB", "#0a1c99"))
cbl_ak <-ggparcoord(test,
           columns = 2:9, groupColumn = 1, showPoints = TRUE, scale = "globalminmax",title="Cerebellum")+theme(plot.title = element_text(size=10),legend.position = "none", axis.text.x = element_text(angle = 45,  hjust = 1))+xlab("")+ylab("expression")+geom_line(size=1.7, alpha=0.7)+scale_color_manual(values = c( "#ABABAB", "#FF0000", "#ABABAB", "#ABABAB", "#ABABAB", "#ABABAB"))
md_ak <-ggparcoord(test,
           columns = 2:9, groupColumn = 1, showPoints = TRUE, scale = "globalminmax",title="MD Thalamus")+theme(plot.title = element_text(size=10),legend.position = "none", axis.text.x = element_text(angle = 45,  hjust = 1))+xlab("")+ylab("expression")+geom_line(size=1.7, alpha=0.7)+scale_color_manual(values = c( "#ABABAB", "#ABABAB", "#ABABAB", "#ffff00", "#ABABAB", "#ABABAB"))

pt2 <- arrangeGrob(boxplotsDist[[2]], top = textGrob("A", x = unit(0, "npc")
         , y   = unit(1, "npc"), just=c("left", "top")))
pt3 <- arrangeGrob(boxplotsDist[[3]], top = textGrob(" ", x = unit(0, "npc")
         , y   = unit(1, "npc"), just=c("left", "top")))
pt4 <- arrangeGrob(boxplotsDist[[4]], top = textGrob(" ", x = unit(0, "npc")
         , y   = unit(1, "npc"), just=c("left", "top")))
pt5 <- arrangeGrob(boxplotsDist[[5]], top = textGrob(" ", x = unit(0, "npc")
         , y   = unit(1, "npc"), just=c("left","top")))
pt6 <- arrangeGrob(boxplotsDist[[6]], top = textGrob(" ", x = unit(0, "npc")
         , y   = unit(1, "npc"), just=c("left","top")))
pt7 <- arrangeGrob(boxplotsDist[[7]], top = textGrob(" ", x = unit(0, "npc")
         , y   = unit(1, "npc"), just=c("left","top")))
pt8 <- arrangeGrob(boxplotsDist[[8]], top = textGrob(" ", x = unit(0, "npc")
         , y   = unit(1, "npc"), just=c("left","top")))
pt9 <- arrangeGrob(boxplotsDist[[9]], top = textGrob(" ", x = unit(0, "npc")
         , y   = unit(1, "npc"), just=c("left","top")))
akeyCBL <- arrangeGrob(cbl_ak, top = textGrob("B", x = unit(0, "npc")
         , y   = unit(1, "npc"), just=c("left","top")))
akMD <- arrangeGrob(md_ak, top = textGrob("C", x = unit(0, "npc")
         , y   = unit(1, "npc"), just=c("left","top")))
akSTR <- arrangeGrob(str_ak, top = textGrob("D", x = unit(0, "npc")
         , y   = unit(1, "npc"), just=c("left","top")))


lay1 <- rbind(c(1,2,3,4,5,6,7,8,9),
             c(NA,10,10,11,11,12,12,NA, NA),
             c(NA,10,10,11,11,12,12,NA,NA))

grid.arrange(pt2, pt3, pt4, pt5, pt6, pt7, pt8, pt9,leg1, akeyCBL, akMD, akSTR,layout_matrix = lay1)
fig3 <- grid.arrange(pt2, pt3, pt4, pt5, pt6, pt7, pt8, pt9,leg1, akeyCBL, akMD, akSTR,layout_matrix = lay1)
#ggsave(file="~/desertsHomo/2.plots/Sestan_AkeyPey_log2_median/fig3_boxplots.pdf", fig3, width = 11.69, height = 8.27, units = "in")
```

#Fig 4 Boxplots + Selected structures
```{r}
figtop4<-ggparcoord(test2,
           columns = 2:9, groupColumn = 1, showPoints = TRUE, scale = "globalminmax",title="C")+theme(plot.title = element_text(size=10),legend.position = "right", axis.text.x = element_text(angle = 45,  hjust = 1))+xlab("")+ylab("expression")+geom_line(size=1.7, alpha=0.7)
leg2 <- get_legend(figtop4)
leg2 <- as_ggplot(leg2)#legend as figure

str_akpey <-ggparcoord(test2,
           columns = 2:9, groupColumn = 1, showPoints = TRUE, scale = "globalminmax",title="Striatum")+theme(plot.title = element_text(size=10),legend.position = "none", axis.text.x = element_text(angle = 45,  hjust = 1))+xlab("")+ylab("expression")+geom_line(size=1.7, alpha=0.7)+scale_color_manual(values = c( "#ABABAB", "#ABABAB", "#ABABAB", "#ABABAB", "#ABABAB", "#0a1c99"))

cbl_akpey <-ggparcoord(test2,
           columns = 2:9, groupColumn = 1, showPoints = TRUE, scale = "globalminmax",title="Cerebellum")+theme(plot.title = element_text(size=10),legend.position = "none", axis.text.x = element_text(angle = 45,  hjust = 1))+xlab("")+ylab("expression")+geom_line(size=1.7, alpha=0.7)+scale_color_manual(values = c( "#ABABAB", "#FF0000", "#ABABAB", "#ABABAB", "#ABABAB", "#ABABAB"))


part2 <- arrangeGrob(boxplotDistakeypey[[2]], top = textGrob("A", x = unit(0, "npc")
         , y   = unit(1, "npc"), just=c("left","top")))
part3 <- arrangeGrob(boxplotDistakeypey[[3]], top = textGrob(" ", x = unit(0, "npc")
         , y   = unit(1, "npc"), just=c("left","top")))
part4 <- arrangeGrob(boxplotDistakeypey[[4]], top = textGrob(" ", x = unit(0, "npc")
         , y   = unit(1, "npc"), just=c("left","top")))
part5 <- arrangeGrob(boxplotDistakeypey[[5]], top = textGrob(" ", x = unit(0, "npc")
         , y   = unit(5, "npc"), just=c("left","top")))
part6 <- arrangeGrob(boxplotDistakeypey[[6]], top = textGrob(" ", x = unit(0, "npc")
         , y   = unit(1, "npc"), just=c("left","top")))
part7 <- arrangeGrob(boxplotDistakeypey[[7]], top = textGrob(" ", x = unit(0, "npc")
         , y   = unit(1, "npc"), just=c("left","top")))
part8 <- arrangeGrob(boxplotDistakeypey[[8]], top = textGrob(" ", x = unit(0, "npc")
         , y   = unit(1, "npc"), just=c("left","top")))
part9 <- arrangeGrob(boxplotDistakeypey[[9]], top = textGrob(" ", x = unit(0, "npc")
         , y   = unit(1, "npc"), just=c("left","top")))

akpeyCBL <- arrangeGrob(cbl_akpey, top = textGrob("B", x = unit(0, "npc")
         , y   = unit(1, "npc"), just=c("left","top")))
akpeySTR <- arrangeGrob(str_akpey, top = textGrob("C", x = unit(0, "npc")
         , y   = unit(1, "npc"), just=c("left","top")))


lay2 <- rbind(c(1,2,3,4,5,6,7,8,9),
             c(NA,NA,10,10,11,11,NA, NA,NA),
             c(NA,NA,10,10,11,11,NA,NA,NA))

grid.arrange(part2, part3, part4, part5, part6, part7, part8, part9,leg2, akpeyCBL,akpeySTR,layout_matrix = lay2)
fig4 <- grid.arrange(part2, part3, part4, part5, part6, part7, part8, part9,leg2, akpeyCBL,akpeySTR,layout_matrix = lay2)
#ggsave(file="~/desertsHomo/2.plots/Sestan_AkeyPey_log2_median/fig4_boxplots.pdf", fig4, width = 11.69, height = 8.27, units = "in")
```


#FIGURE #S1
```{r}
preli1 <- read.csv(file="~/desertsHomo/1.data/median_tables/median_Akey_Sestan.csv")
preli1 <- as_tibble(preli1) #AKEY

preli2 <- read.csv(file="~/desertsHomo/1.data/median_tables/median_AkeyPey_Sestan.csv")
preli2 <- as_tibble(preli2) #AKEYpey

preli3 <- read.csv(file="~/desertsHomo/1.data/median_tables/median_filtered_rawSestan.csv")
preli3 <- as_tibble(preli3) #rawfiltered

colnames(preli1) <- c("Structure", "Fetal_1", "Fetal_2", "Fetal_3", "Birth/Ifan", "Infan/Child", "Child", "Adolescence", "Adult")
#PLOT
levels(colnames(preli1)) <- c("Structure", "Fetal_1", "Fetal_2", "Fetal_3", "Birth/Inf", "Inf/Child", "Child", "Adolescence", "Adult")

colnames(preli2) <- c("Structure", "Fetal_1", "Fetal_2", "Fetal_3", "Birth/Ifan", "Infan/Child", "Child", "Adolescence", "Adult")
#PLOT
levels(colnames(preli2)) <- c("Structure", "Fetal_1", "Fetal_2", "Fetal_3", "Birth/Inf", "Inf/Child", "Child", "Adolescence", "Adult")

colnames(preli3) <- c("Structure", "Fetal_1", "Fetal_2", "Fetal_3", "Birth/Ifan", "Infan/Child", "Child", "Adolescence", "Adult")
#PLOT
levels(colnames(preli3)) <- c("Structure", "Fetal_1", "Fetal_2", "Fetal_3", "Birth/Inf", "Inf/Child", "Child", "Adolescence", "Adult")

preli1$dataset <- c("Deserts")
preli2$dataset <- c("Deserts and Pos. Sel.")
preli3$dataset <- c("Global profile")

tot_pl <-rbind(preli1, preli2, preli3)
tot_pl <- as_tibble(tot_pl)
tot_pl <- tot_pl %>% mutate(dataset=as.character(dataset))

levels(tot_pl$dataset) <-  c("Deserts", "Deserts and Pos. Sel.", "Global profile")


n <-ggparcoord(tot_pl,
columns = 2:9, groupColumn = 1, showPoints = TRUE, scale = "globalminmax", mapping=aes(color=as.factor(dataset)))+theme(plot.title = element_text(size=10), axis.text.x = element_text(angle = 45,  hjust = 1))+xlab("")+ylab("expression")+labs(color="Dataset")

n + facet_wrap(~Structure)+scale_color_discrete(name="Dataset",labels=unique(tot_pl$dataset))

an <- n + facet_wrap(~Structure)+scale_color_discrete(name="Dataset",labels=unique(tot_pl$dataset))
#ggsave(file="~/desertsHomo/2.plots/Sestan_raw_filtered_median2_trajectories/Sestan_temporal_Structures_filteredRawmedian2.pdf", an, width = 11.69, height = 8.27, units = "in")
```

#FIGURE #S2
```{r}
ap <- finalakeySestan1 %>% select(both$hgnc_symbol|Label)
tes2 <- melt(ap, id=c("Label"))
tes2_mean <- tes2 %>%
  group_by(Label,variable) %>%
  dplyr::summarise_all(median, na.rm=TRUE)

tes2_mean <- tes2_mean %>% 
  separate(Label, c("Structure","Window"))
prelites2 <- pivot_wider(tes2_mean, names_from = Window, values_from = value)

#
prelites2[,11] <- NULL
prelites2 <- prelites2[complete.cases(prelites2), ]

#levels(prelites2$Genename) <- unique(prelites2$Genename)

colnames(prelites2) <- c("Structure", "Genename", "Fetal_1", "Fetal_2", "Fetal_3", "Birth/Ifan", "Infan/Child", "Child", "Adolescence", "Adult")
#PLOT
levels(colnames(prelites2)) <- c("Structure", "Genename", "Fetal_1", "Fetal_2", "Fetal_3", "Birth/Inf", "Inf/Child", "Child", "Adolescence", "Adult")

n <-ggparcoord(prelites2,
columns = 3:10, groupColumn = 2, showPoints = TRUE, scale = "globalminmax", mapping=aes(color=factor(Structure)))+xlab("")+ylab("expression")+theme(axis.text.x = element_text(angle = 45,hjust = 1))
j1<-n + facet_wrap(~Genename)+scale_color_discrete(name="Structure",labels=unique(prelites2$Structure))
#ggsave(file="~/desertsHomo/2.plots/Sestan_AkeyPey_log2_median/GenesAkeyPey_log2_median_perGenes_1.pdf", j1, width = 11.69, height = 8.27, units = "in")

n1 <-ggparcoord(prelites2,
columns = 3:10, groupColumn = 1, showPoints = TRUE, scale = "globalminmax",title="Genes in Deserts and Pos Sel", mapping=aes(color=factor(Genename)))+xlab("")+ylab("expression")
j2 <- n1 + facet_wrap(~Structure)+scale_color_discrete(name="Genes",labels=unique(prelites2$Genename))+theme(axis.text.x = element_text(angle = 45,hjust = 1))
#ggsave(file="~/desertsHomo/2.plots/Sestan_AkeyPey_log2_median/GenesAkeyPey_log2_median_perGenes_2.pdf", j2, width = 11.69, height = 8.27, units = "in")
```


